### PR TITLE
Fix docker app debug test

### LINKED ISSUE #101: Fix Docker remote debugging test
The docker based app deployer test in debug mode requires special handling with the following changes:

1) In this test case, our test assertion to check the deployment status requires different approach for maven vs docker as maven goes through the deployer status deploying -> deployed whereas the docker resource deployment checks for the container process running status hence, it goes directly to `deployed` status. 

For docker based app, explicitly verify the docker container's log output on debug suspend message to verify the correctness of this test case.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Make sure to pass debug options as JAVA_OPTS
 - Verify the docker container log for the debug listener message

Resolves #101

### CODE REVIEW

**Comment by ilayaperumalg** [comment]
This PR requires changes from https://github.com/spring-cloud/spring-cloud-deployer/pull/247

**Comment by ilayaperumalg** [comment]
To build the docker image: `./mvnw -pl spring-cloud-deployer-spi-test-app clean package dockerfile:build verify`

File: spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/AbstractLocalDeployerSupport.java [hunk 1]
```diff
@@ -186,7 +186,8 @@ protected ProcessBuilder buildProcessBuilder(AppDeploymentRequest request, Map<S
 
 			if(request.getResource() instanceof DockerResource) {
 				builder.command().add(2, "-e");
-				builder.command().add(3, "JAVA_TOOL_OPTIONS=\""+ debugInstruction + "\"");			}
+				builder.command().add(3, "JAVA_TOOL_OPTIONS="+ debugInstruction);
```
Comment (markpollack) [inline spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/AbstractLocalDeployerSupport.java]: This is not adding support for passing the deployer property `spring.cloud.deployer.local.javaOpts`.  It appears to be fixing something wrt to debug support, which afaik, we had tests for that were working.

**Comment by markpollack** [comment]
squashed and merged as 6eee669f4088af4ff1ca9a9cd406230d8fcfd284

**Pull request closed without merging**